src/DataFramesMeta.jl

```diff
@@ -104,44 +104,67 @@ function fun_to_vec(kw::Expr; nolhs::Bool = false, gensym_names::Bool = false)
     # nolhs: f(:x) where f returns a Table
     # !nolhs, y = g(:x)
     if kw.head === :(=) || kw.head === :kw || nolhs
-        membernames = Dict{Any, Symbol}()
-        if nolhs
-            # act on f(:x)
-            body = replace_syms!(kw, membernames)
+        # kw.args[2] is RHS of kw
+        # this checks if the RHS is of form f(:x, :y),
+        # in wihch case we generate [:x, :y] => f => ...
+        # to avoid compilation cost of anonymous function.
+        #
+        # Restrict version due to https://github.com/JuliaLang/julia/pull/37583
+        if VERSION > v"1.6.0-DEV" &&
+            nolhs == false &&
+            headmatch(kw.args[2], :call) &&
+            all(s -> s isa QuoteNode, kw.args[2].args[2:end])
+
+            source = Expr(:vect, kw.args[2].args[2:end]...)
+            fun = kw.args[2].args[1]
         else
-            # act on g(:x)
-            body = replace_syms!(kw.args[2], membernames)
-        end
+            membernames = Dict{Any, Symbol}()
+            if nolhs
+                # act on f(:x)
+                body = replace_syms!(kw, membernames)
+            else
+                # act on g(:x)
+                body = replace_syms!(kw.args[2], membernames)
+            end
 
-        source = Expr(:vect, keys(membernames)...)
+            source = Expr(:vect, keys(membernames)...)
+            inputargs = Expr(:tuple, values(membernames)...)
+            fun = quote
+                (@nospecialize args...) -> begin
```

Comment:
but if this is the case and we removed the `@nospecialize` here how would the benchmarks be affected?

Reply:
They are completely unchanged. 

I think it would be useful to try and find scenarios where they *are* affected. I will do that this weekend.

Reply:
That is what I was wanting to understand 😄